pkg/resources/fluentd/fluentd.go

```diff
@@ -48,19 +52,26 @@ var labelSelector = map[string]string{
 type Reconciler struct {
 	Logging *v1beta1.Logging
 	*k8sutil.GenericResourceReconciler
-	config *string
+	config  *string
+	secrets *secret.MountSecrets
 }
 
-func New(client client.Client, log logr.Logger, logging *v1beta1.Logging, config *string) *Reconciler {
+func New(client client.Client, log logr.Logger, logging *v1beta1.Logging, config *string, secrets *secret.MountSecrets) *Reconciler {
 	return &Reconciler{
 		Logging:                   logging,
 		GenericResourceReconciler: k8sutil.NewReconciler(client, log),
 		config:                    config,
+		secrets:                   secrets,
 	}
 }
 
 // Reconcile reconciles the fluentd resource
 func (r *Reconciler) Reconcile() (*reconcile.Result, error) {
+	// Prepare output secret
+	err := r.ReconcileResource(r.outputSecret(r.secrets, OutputSecretPath))
+	if err != nil {
+		return nil, errors.WrapIf(err, "failed to reconcile resource")
+	}
```

Comment:
is this necessary to do before the configcheck, or could we do it after that where all the other resources are reconciled?